diff --git a/ktsctl/build.rs b/ktsctl/build.rs
index cbc909f..feffeba 100644
--- a/ktsctl/build.rs
+++ b/ktsctl/build.rs
@@ -1,12 +1,5 @@
-use std::process::Command;
-
 fn main() {
-  let output = Command::new("git")
-    .args(["rev-parse", "--short", "HEAD"])
-    .output()
-    .unwrap();
-
-  let key = "GIT_HEAD";
-  let value = String::from_utf8(output.stdout).unwrap();
-  println!("cargo:rustc-env={key}={value}");
+    let key = "GIT_HEAD";
+    let value = "b390d1f";
+    println!("cargo:rustc-env={}={}", key, value);
 }
